.github/workflows/TestSuites.yaml

```diff
@@ -107,19 +118,35 @@ jobs:
           entrypoint.sh cmake --build . --parallel 1 
       - name: Run tests ${{ matrix.part }}/2
         run: |
-          # derive whether or not to run with MPI.
-          if [ "${{ matrix.config[1] }}" = 'mpicc' ]
-          then
-            cd build
-            # only run MPI tests. Apparently mpi in the container doesn't need `--oversubscribe`
-            ctest --output-on-failure -R MPIParallelAutoPasTests
-            cd examples
-            # exclude examples that do not support MPI 
-            ctest --output-on-failure -C checkExamples -E 'sph-((main.test)|(diagram.*))'
+          # For Multi-Site Molecule Md-flexible builds, only run tests for md-flexible.
+          if [ "${{ matrix.md-flex-mode }}" = 0 ]
+            then
+            # derive whether or not to run with MPI.
+            if [ "${{ matrix.config[1] }}" = 'mpicc' ]
```

Comment:
I agree, it's an ugly tree. Your suggested example looks like a very nice solution if we were to give certain builds "tags" and run certain jobs based on which "tags" are used. In particular, for CI's the leaves of the tree would mix and match which small jobs to run. However for our CI tree all the leaves result in entirely unique leaves unless we were to get clever about fusing the MPI example tests and non-MPI example tests into one line of code (which I'm not sure is worth it). So I would suggest a simpler approach:

Would you be fine if I were to largely keep the CI here the same, but instead of having a tree of `if` statements, we have 4 conditionals like `if [ "${{ matrix.md-flex-mode }}" = 0 ] and [ "${{ matrix.config[1] }}" = 'mpicc' ]`?

Reply:
sure give it a try